src/assignmentNotification.js

```diff
@@ -19,25 +19,13 @@ AssignmentNotification.prototype.send = function() {
   // the assignment notification from the analytics write success we can
   // bring this down to 1 HTTP request
 
-  var firstPersist = this._persistAssignment();
+  const first = this._persistAssignment();
 
-  var secondPersist = $.Deferred();
-  this._visitor.analytics.trackAssignment(
-    this._visitor.getId(),
-    this._assignment,
-    function(success) {
-      var promise = this._persistAssignment(success ? 'success' : 'failure');
-      promise.always(function() {
-        if (promise.state() === 'resolved') {
-          secondPersist.resolve();
-        } else {
-          secondPersist.reject();
-        }
-      });
-    }.bind(this)
-  );
+  const second = this._visitor.analytics
+    .trackAssignment(this._visitor.getId(), this._assignment)
+    .then(success => this._persistAssignment(success ? 'success' : 'failure'));
 
-  return $.when(firstPersist, secondPersist);
+  return Promise.all([first, second]);
```

Comment:
We're going to have merge conflicts here with my PR.

Reply:
I'll hold off on this until we get yours in since that's a smaller diff with `master`.